src/pudl/metadata/codes.py

```diff
@@ -53,6 +134,424 @@
+                ("SM", "smoot_co", "Smoot Company"),
+                ("ST", "sterling", "Sterling"),
+                ("SW", "siemens_westinghouse", "Siemens-Westinghouse"),
+                ("TC", "turbosonic", "Turbosonic"),
+                ("TEC", "thermal_equipment_corp", "Thermal Equipment Corporation"),
+                ("TH", "thyssen_cea", "Thyssen/CEA"),
+                ("TK", "turbotak", "Turbotak"),
+                ("TM", "tampella_power_corp", "Tampella Power Corporation"),
+                ("TP", "tempala_power", "Tempala Power"),
+                ("TS", "toshiba", "Toshiba"),
+                ("UE", "utility_engineering", "Utility Engineering"),
+                ("UM", "united_mcgill", "United McGill"),
+                ("UO", "universal_oil_products", "Universal Oil Products"),
+                ("VO", "vogt", "Vogt Machine Company/Vogt Power"),
+                (
+                    "WAP",
+                    "wheelabrator_air_pollution_control",
+                    "Wheelabrator Air Pollution Control",
+                ),
+                ("WE", "westinghouse", "Westinghouse"),
+                ("WG", "weigl_engineering", "Weigl Engineering"),
+                ("WI", "wickes", "Wickes"),
+                ("ZC", "zeeco", "Zeeco"),
+                ("ZN", "zurn", "Zurn"),
+            ],
+        ).convert_dtypes(),
+        "code_fixes": {},
+        "ignored_codes": ["NA", "IN"],
+    },
+    "firing_type_codes_eia": {
+        "df": pd.DataFrame(
+            columns=["code", "label", "description"],
+            data=[
+                ("CB", "cell_burner", "A boiler with a cell burner."),
+                ("CY", "cyclone_firing", "A cyclone-fired boiler."),
+                ("DB", "duct_burner", "A boiler with a duct burner."),
+                (
+                    "FB",
+                    "fluidized_bed_firing",
+                    "A boiler with fluidized bed firing (circulating fluidized bed, bubbling fluidized bed).",
+                ),
+                (
+                    "SS",
+                    "stoker",
+                    "A stoker-fired boiler (spreader, vibrating Gate, slinger).",
+                ),
+                (
+                    "TF",
+                    "tangential_firing",
+                    "A boiler with tangential, concentric or corner firing.",
+                ),
+                ("VF", "vertical_firing", "A vertical or arch-fired boiler."),
+                (
+                    "WF",
+                    "wall_fired",
+                    "A wall-fired boiler (opposed wall, rear wall, front wall, side wall).",
+                ),
+                ("OT", "other", "Other: specify in Schedule 7."),
+            ],
+        ).convert_dtypes(),
+        "code_fixes": {
+            "AF": "VF",
+            "CF": "TF",
+            "FF": "WF",
+            "OF": "WF",
+            "RF": "WF",
+            "SF": "WF",
```

Comment:
Did these codes get more or less granular over time? If they got more granular over time, does it seem like we could backfill with the more detailed codes?

Reply:
The codes get less granular over time.

Reply:
I might be misunderstanding your summary from the top level PR comment, but it seemed like in some cases you were merging codes, and in others you were keeping both the generic and granular codes. If that's true, what's your thinking there? Are you just trying to make sure that the data reflects the most recent reporting structure?

Reply:
Where codes are getting less granular over time (i.e. here, where the newer code WF includes older codes FF, OF, RF, SF), I have merged them to reflect the most recent reporting structure. Where older data is less specific (e.g. just specifies a dry scrubber) and I'm not able to clearly identify what more specific subcategory the record could be matched to, I add the older more generic categories to the dictionary and make a note in the field metadata.